source/extensions/formatter/req_without_query/req_without_query.cc

```diff
@@ -0,0 +1,90 @@
+#include "extensions/formatter/req_without_query/req_without_query.h"
+
+#include <string>
+
+#include "common/protobuf/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Formatter {
+
+void stripQueryString(std::string& path) {
+  const size_t query_pos = path.find('?');
+  path = std::string(path.data(), query_pos != path.npos ? query_pos : path.size());
```

Comment:
Let's use Http::Utility::Url, e.g.:

https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/csrf/csrf_filter.cc#L40

instead of manually parsing...

Reply:
I checked `Http::Utility::Url` but I did not find a method for returning path without query string.  There seemed to be very few methods there in the [class](https://github.com/envoyproxy/envoy/blob/7d426641e044491ed13b0b45afb25604c323fbc9/source/common/http/utility.h#L135) in general. On the other hand, there seems to be several stand-alone utility functions for manipulating URLs as plain string, like [Utility::parseAndDecodeQueryString()](https://github.com/envoyproxy/envoy/blob/7d426641e044491ed13b0b45afb25604c323fbc9/source/common/http/utility.cc#L331) that seem to mostly use `url.find('?')` to separate the query string from path.

Did you mean that I should create method new method in `Http::Utility::Url` class for returning path without query string?  Or alternatively maybe move this simple `stripQueryString()` to the collection of utility functions `common/http/utility.cc`?

Reply:
+1 for `stripQueryString()` in `Http::Utility::Url`.

Reply:
I can do that but `Http::Utility::Url` parser seems [relatively complicated](https://github.com/envoyproxy/envoy/blob/118f66e87bae3961c5726ad46239a85cdadc316f/source/common/http/utility.cc#L256), even to the extent that it uses nodejs http-parser library.  On the other hand there were other utility functions in `Http::Utility` namespace that did simple string searches for path, like [Utility::findQueryStringStart()](https://github.com/envoyproxy/envoy/blob/118f66e87bae3961c5726ad46239a85cdadc316f/source/common/http/utility.cc#L373).  I wonder if it would be OK to follow the latter example, so I've added `Utility::stripQueryString()` to this PR for now but if you prefer `Http::Utility::Url` I'll dig into nodejs http-parser library.